samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -94,6 +95,36 @@ public ZkKeyBuilder getKeyBuilder() {
     return keyBuilder;
   }
 
+  public static class ProcessorData {
```

Comment:
Well, that what it was before. But now, I actually think your previous comment about abstracting the data format is a good idea event if at this point it may be just a simple string.

And now, come to think of it, hostname will be very usefully when debugging a really distributed app. So I am going to leave it there, and add it to some logging.